fastapi/_compat.py

```diff
@@ -230,11 +230,7 @@ def get_definitions(
         return field_mapping, definitions  # type: ignore[return-value]
 
     def is_scalar_field(field: ModelField) -> bool:
-        from fastapi import params
-
-        return field_annotation_is_scalar(
-            field.field_info.annotation
-        ) and not isinstance(field.field_info, params.Body)
```

Comment:
This check was missing from v1 side. Lifted the `params.Body` check out from this "scalar field check" into `is_body_param` check which is more appropriate for checking the `params.Body` case.